packages/p2p/src/components/advertiser-page/advertiser-page.jsx

```diff
@@ -60,18 +39,17 @@ const AdvertiserPage = observer(() => {
         total_orders_count,
     } = advertiser_page_store.advertiser_info;
 
-    const avg_release_time_in_minutes = release_time_avg > 60 ? Math.round(release_time_avg / 60) : '< 1';
-    const Form = general_store.nickname ? BuySellForm : NicknameForm;
-
-    const Row = row_props => <RowComponent {...row_props} showAdPopup={advertiser_page_store.showAdPopup} />;
-
     React.useEffect(() => {
         advertiser_page_store.onMount();
-    }, []);
 
-    React.useEffect(() => {
-        advertiser_page_store.onTabChange();
-    }, [advertiser_page_store.active_index]);
+        return reaction(
+            () => advertiser_page_store.active_index,
+            () => advertiser_page_store.onTabChange(),
+            { fireImmediately: true }
```

Comment:
Why are we adding this, when `useEffect` is called on mount?

Reply:
A MobX reaction normally doesn't execute until the next observable change, so in this case we'd only register the reaction for any future changes. The `fireImmediately` option will trigger it 🥁... immediately.